renku/core/models/datasets.py

```diff
@@ -744,6 +727,39 @@ class Meta:
     same_as = fields.Nested(schema.sameAs, UrlSchema, missing=None)
     short_name = fields.String(schema.alternateName)
 
+    @pre_load
+    def fix_files_context(self, data, **kwargs):
```

Comment:
Is this needed all the time or just initially for a migration? If the latter, could we add the `pre_load` code inside a relevant migration (copy/subclass the schema there) instead of here?

Reply:
Actually, this one is not easy to get rid of; it is required for all project versions (2 ATM). Ideally we should refactor migrations to be independent from the core models so that these won't be required. I've created a followup story: https://github.com/SwissDataScienceCenter/renku-python/issues/1340.